clients/admin-ui/src/features/integrations/configure-monitor/ConfigureMonitorModal.tsx

```diff
@@ -26,36 +32,71 @@ const ConfigureMonitorModal = ({
   onAdvance: (m: MonitorConfig) => void;
   integration: ConnectionConfigurationResponse;
   integrationOption: ConnectionSystemTypeMap;
-}) => (
-  <AddModal
-    title={
-      monitor?.name
-        ? `Configure ${monitor.name}`
-        : "Configure discovery monitor"
+}) => {
+  const [putMonitorMutationTrigger, { isLoading: isSubmitting }] =
+    usePutDiscoveryMonitorMutation();
+
+  const { data: databases } = useGetDatabasesByConnectionQuery({
```

Comment:
Perfect. Having the data fetching higher up in the component hierarchy is very useful.